packages/bundle/src/FullReactWebChat.js

```diff
@@ -1,93 +1,47 @@
+import React, { useEffect } from 'react';
 
-import AdaptiveCardsContext from './adaptiveCards/AdaptiveCardsContext';
-import createAdaptiveCardsAttachmentMiddleware from './adaptiveCards/createAdaptiveCardMiddleware';
-import createDefaultAdaptiveCardHostConfig from './adaptiveCards/Styles/adaptiveCardHostConfig';
-import createStyleSet from './adaptiveCards/Styles/createStyleSetWithAdaptiveCards';
-import defaultRenderMarkdown from './renderMarkdown';
+import AdaptiveCardsComposer from './adaptiveCards/AdaptiveCardsComposer';
+import useComposerProps from './useComposerProps';
 
 // Add additional props to <WebChat>, so it support additional features
-const FullReactWebChat = ({
-  adaptiveCardHostConfig,
-  adaptiveCardsHostConfig,
-  adaptiveCardsPackage,
-  attachmentMiddleware,
-  renderMarkdown,
-  styleOptions,
-  styleSet,
-  ...otherProps
-}) => {
+const FullReactWebChat = props => {
+  const { adaptiveCardHostConfig, adaptiveCardsHostConfig, adaptiveCardsPackage, ...otherProps } = props;
+
   useEffect(() => {
     adaptiveCardHostConfig &&
       console.warn(
         'Web Chat: "adaptiveCardHostConfig" is deprecated. Please use "adaptiveCardsHostConfig" instead. "adaptiveCardHostConfig" will be removed on or after 2022-01-01.'
       );
   }, [adaptiveCardHostConfig]);
 
-  const patchedStyleSet = useMemo(() => styleSet || createStyleSet(styleOptions), [styleOptions, styleSet]);
-  const { options: patchedStyleOptions } = patchedStyleSet;
-
-  const patchedAdaptiveCardsHostConfig = useMemo(
-    () => adaptiveCardsHostConfig || adaptiveCardHostConfig || createDefaultAdaptiveCardHostConfig(patchedStyleOptions),
-    [adaptiveCardHostConfig, adaptiveCardsHostConfig, patchedStyleOptions]
-  );
-
-  const patchedAdaptiveCardsPackage = useMemo(() => adaptiveCardsPackage || defaultAdaptiveCardsPackage, [
-    adaptiveCardsPackage
-  ]);
-
-  const patchedRenderMarkdown = useMemo(
-    () => renderMarkdown || (markdown => defaultRenderMarkdown(markdown, patchedStyleOptions)),
-    [renderMarkdown, patchedStyleOptions]
-  );
-
-  const patchedAttachmentMiddleware = useMemo(
-    () => concatMiddleware(attachmentMiddleware, createAdaptiveCardsAttachmentMiddleware()),
-    [attachmentMiddleware]
-  );
-
-  const adaptiveCardsContext = useMemo(
-    () => ({
-      adaptiveCardsPackage: patchedAdaptiveCardsPackage,
-      hostConfig: patchedAdaptiveCardsHostConfig
-    }),
-    [patchedAdaptiveCardsHostConfig, patchedAdaptiveCardsPackage]
-  );
+  const composerProps = useComposerProps(props);
 
   return (
-    <AdaptiveCardsContext.Provider value={adaptiveCardsContext}>
-      <BasicWebChat
-        attachmentMiddleware={patchedAttachmentMiddleware}
-        renderMarkdown={patchedRenderMarkdown}
-        styleOptions={styleOptions}
-        styleSet={patchedStyleSet}
-        {...otherProps}
-      />
-    </AdaptiveCardsContext.Provider>
+    <AdaptiveCardsComposer
+      adaptiveCardsHostConfig={adaptiveCardsHostConfig}
```

Comment:
The custom host config tests are failing because `adaptiveCardHostConfig` isn't being passed to the `AdaptiveCardComposer` correctly.

```suggestion
      adaptiveCardsHostConfig={adaptiveCardHostConfig || adaptiveCardsHostConfig}
```

Reply:
Thanks! How'd I miss this one? 🤦🏻‍♂️